tests/Unit/Evolution/Systems/GrMhd/GhValenciaDivClean/Test_TimeDerivativeTerms.cpp

```diff
@@ -0,0 +1,233 @@
+                                    gradient_variables,
+                                    all_valencia_argument_variables);
+
+  // compute the stress energy for the expected variables
+  grmhd::GhValenciaDivClean::trace_reversed_stress_energy(
+      make_not_null(
+          &get<grmhd::GhValenciaDivClean::Tags::TraceReversedStressEnergy>(
+              temp_variables)),
+      make_not_null(&get<grmhd::GhValenciaDivClean::Tags::FourVelocityOneForm>(
+          temp_variables)),
+      make_not_null(
+          &get<grmhd::GhValenciaDivClean::Tags::ComovingMagneticFieldOneForm>(
+              temp_variables)),
+      tuples::get<hydro::Tags::RestMassDensity<DataVector>>(arg_variables),
+      tuples::get<hydro::Tags::SpecificEnthalpy<DataVector>>(arg_variables),
+      get<grmhd::GhValenciaDivClean::detail::ValenciaTempTag<
+          hydro::Tags::SpatialVelocityOneForm<DataVector, 3, Frame::Inertial>>>(
+          temp_variables),
+      get<grmhd::GhValenciaDivClean::detail::ValenciaTempTag<
+          hydro::Tags::MagneticFieldOneForm<DataVector, 3>>>(temp_variables),
+      get<grmhd::GhValenciaDivClean::detail::ValenciaTempTag<
+          hydro::Tags::MagneticFieldSquared<DataVector>>>(temp_variables),
+      get<grmhd::GhValenciaDivClean::detail::ValenciaTempTag<
+          hydro::Tags::MagneticFieldDotSpatialVelocity<DataVector>>>(
+          temp_variables),
+      tuples::get<hydro::Tags::LorentzFactor<DataVector>>(arg_variables),
+      get<grmhd::GhValenciaDivClean::detail::ValenciaTempTag<
+          typename grmhd::ValenciaDivClean::TimeDerivativeTerms::
+              OneOverLorentzFactorSquared>>(temp_variables),
+      tuples::get<hydro::Tags::Pressure<DataVector>>(arg_variables),
+      tuples::get<gr::Tags::SpacetimeMetric<3_st>>(arg_variables),
+      get<gr::Tags::Shift<3_st>>(temp_variables),
+      get<gr::Tags::Lapse<>>(temp_variables));
+  // apply the correction to dt pi for the expected variables
+  grmhd::GhValenciaDivClean::add_stress_energy_term_to_dt_pi(
+      make_not_null(&get<::Tags::dt<GeneralizedHarmonic::Tags::Pi<3_st>>>(
+          expected_dt_variables)),
+      get<grmhd::GhValenciaDivClean::Tags::TraceReversedStressEnergy>(
+          temp_variables),
+      get<gr::Tags::Lapse<>>(temp_variables));
+
+  ComputeVolumeTimeDerivativeTermsHelper<
+      grmhd::GhValenciaDivClean::TimeDerivativeTerms<equation_of_state_type>,
+      3_st, tmpl::append<gh_variables_tags, valencia_variables_tags>,
+      typename flux_variables_type::tags_list,
+      typename temp_variables_type::tags_list,
+      typename gradient_variables_type::tags_list,
+      typename arg_variables_type::tags_list>::
+      apply(make_not_null(&dt_variables), make_not_null(&flux_variables),
+            make_not_null(&temp_variables), gradient_variables, arg_variables);
+
+  CHECK_VARIABLES_APPROX(dt_variables, expected_dt_variables);
+  CHECK_VARIABLES_APPROX(flux_variables, expected_flux_variables);
```

Comment:
you might be able to have an `expected_temp_tags` here without too much work, up to you :)